cancelei/budget-in-rails
# Implement Design and Interaction Features for Budget-in-Rails APP

**Design Implementation**

Colors, Typography, and Layout: The UI has been updated to follow the provided design guidelines. This includes the application of color schemes, font faces, sizes, and weights, as well as the layout specifications for spacing and element composition.

Design Credit: As per the Creative Commons license, credit to Gregoire Vella on Behance has been appropriately added to the README file.

**User Interactions**
Splash Screen with Sign Up and Log In Pages: Users can now register with their full name, email, and password, and log in using their credentials. Access control has been put in place to prevent unauthorized users from accessing certain pages.
Home Page (Categories Page): Upon login, users are directed to the categories page where they can view each category with its name, icon, and total transaction amount.

Transactions Page: Transactions for a specific category are listed in descending order by date. Users can view the total amount per category and navigate to add new transactions.
"Add a New Category" Page: A form has been created to allow users to add new categories, with fields for name and icon.
"Add a New Transaction" Page: A form has been created for adding new transactions, with mandatory fields for name, amount, and category.

**Testing Requirements**
Comprehensive unit and integration tests have been written and passed for the most crucial components of the RoR application.

**Technical Requirements**
Database: Postgres has been integrated and configured as the primary database for the application.
Authentication: Devise is now handling user authentication.
Input Validation: All user inputs are now validated to prevent potential security breaches.
View Template Engine: The chosen template engine for views is [.erb/.slim/.haml] (select as appropriate).
Deployment: The application has been deployed and is now accessible online.

**Database Schema**
The database schema has been updated to reflect the required structure, with modifications to the "Entity" name to suit the application's theme and avoid conflicts with ActiveRecord's reserved names.

**How to Test This Feature**
Navigate to the splash screen and verify the presence of the app name and navigation links.
Register a new user and attempt to log in with the new credentials.
Once logged in, verify that the categories page loads with the correct information.
Navigate to the transactions page by selecting a category and confirm the transactions are listed correctly.
Use the "add a new category" feature and ensure the form behaves as expected.
Add a new transaction and check for correct functionality and navigation.

**Additional Notes**

Some pages were created without provided designs; these were implemented by following the design guidelines of the other pages.
The "Transaction" entity name has been changed to "Operation / Operations" avoid naming conflicts with ActiveRecord.
Please review the changes and provide your feedback.
